jobsapp/api/views/users.py

```diff
@@ -1,18 +1,18 @@
-from rest_framework.generics import CreateAPIView, ListAPIView, RetrieveUpdateDestroyAPIView
-from rest_framework.permissions import AllowAny
+from rest_framework.generics import CreateAPIView, ListCreateAPIView, RetrieveUpdateDestroyAPIView
 
 from ...models import User
+from ..permissions import IsSelfOrReadOnly
 from ..serializers import UserSerializer
 
 
-class UsersViewsList(ListAPIView):
+class UsersViewsList(ListCreateAPIView):
     serializer_class = UserSerializer
     queryset = User.objects.all()
-    permission_classes = [AllowAny]
+    permission_classes = [IsSelfOrReadOnly]
```

Comment:
It must be `permission_classes = [AllowAny]` otherwise we aren't allowing new users to be created.